lib/Bitcode/Reader/BitstreamReader.cpp

```diff
@@ -210,9 +292,55 @@ unsigned BitstreamCursor::readRecord(unsigned AbbrevID,
           EltEnc.getEncoding() == BitCodeAbbrevOp::Blob)
         report_fatal_error("Array element type can't be an Array or a Blob");
 
+#if 1 // HLSL Change
+      // Read all the elements a little faster.
+      {
+        BitCodeAbbrevOp::Encoding enc = EltEnc.getEncoding();
+        uint64_t encData = 0;
+        if (EltEnc.hasEncodingData())
+          encData = EltEnc.getEncodingData();
+        unsigned size = (unsigned)encData;
+        if (Uint8Vals) {
+          auto &Vals = *Uint8Vals;
```

Comment:
Ok, it was confusing that &Vals hides the parameter &Vals.  It seems unnecessary, since you could just use `(*Uint8Vals)` below directly instead.